server/src/main/java/org/opensearch/wlm/QueryGroupService.java

```diff
@@ -331,7 +331,7 @@ public Set<QueryGroup> getDeletedQueryGroups() {
     public boolean shouldSBPHandle(Task t) {
         QueryGroupTask task = (QueryGroupTask) t;
         boolean isInvalidQueryGroupTask = true;
-        if (!task.getQueryGroupId().equals(QueryGroupTask.DEFAULT_QUERY_GROUP_ID_SUPPLIER.get())) {
+        if (task.isQueryGroupSet() && !task.getQueryGroupId().equals(QueryGroupTask.DEFAULT_QUERY_GROUP_ID_SUPPLIER.get())) {
```

Comment:
Can we prefer `task.getQueryGroupId().equals(QueryGroupTask.DEFAULT_QUERY_GROUP_ID_SUPPLIER.get())` over `task.getQueryGroupId().equals(QueryGroupTask.DEFAULT_QUERY_GROUP_ID_SUPPLIER.get())` to avoid any NullPointerException. Else, we should explicitly check for `task.getQueryGroupId() != null`

Reply:
Can we make similar change at other places to avoid NPE?

Reply:
We have done that in other places, `task.isQueryGroupSet()` is better because if the queryGroupId is not set then task.getQueryGroupId will generate the warn log.

Reply:
I think I have covered all the places where getQueryGroupId is called but can double check it

Reply:
Sorry meant to say `QueryGroupTask.DEFAULT_QUERY_GROUP_ID_SUPPLIER.get().equals(task.getQueryGroupId())` over the current one!